kinder/ci/tools/update-workflows/templates/workflows/rootless-tasks.yaml

```diff
@@ -0,0 +1,279 @@
+# IMPORTANT! this workflow is imported by rootless-* workflows.
+version: 1
+summary: |
+  This workflow implements a sequence of tasks used test the proper functioning
+  of kubeadm with the control-plane running as non-root.
+vars:
+  # vars defines default values for variable used by tasks in this workflow;
+  # those values might be overridden when importing this files.
+  kubernetesVersion: v1.13.5
+  controlPlaneNodes: 3
+  workerNodes: 2
+  baseImage: kindest/base:v20191105-ee880e9b # has containerd
+  image: kindest/node:test
+  clusterName: kinder-regular
+  kubeadmVerbosity: 6
+tasks:
+- name: pull-base-image
+  description: |
+    pulls kindest/base image with docker in docker and all the prerequisites necessary for running kind(er)
+  cmd: docker
+  args:
+    - pull
+    - "{{ .vars.baseImage }}"
+- name: add-kubernetes-versions
+  description: |
+    creates a node-image-variant by adding a Kubernetes version
+  cmd: kinder
+  args:
+    - build
+    - node-image-variant
+    - --base-image={{ .vars.baseImage }}
+    - --image={{ .vars.image }}
+    - --with-init-artifacts={{ .vars.kubernetesVersion }}
+    - --loglevel=debug
+  timeout: 15m
+- name: create-cluster
+  description: |
+    create a set of nodes ready for hosting the Kubernetes cluster
+  cmd: kinder
+  args:
+    - create
+    - cluster
+    - --name={{ .vars.clusterName }}
+    - --image={{ .vars.image }}
+    - --control-plane-nodes={{ .vars.controlPlaneNodes }}
+    - --worker-nodes={{ .vars.workerNodes }}
+    - --loglevel=debug
+  timeout: 5m
+- name: prepare verify-rootless.sh script
+  cmd: /bin/sh
+  args:
+    - -c
+    - |
+      cat <<EOF >/tmp/verify-rootless.sh
+      #!/usr/bin/env bash
+      res=0
+      users=("kubeadm-kas" "kubeadm-ks" "kubeadm-kcm" "kubeadm-etcd")
+      for d in ${users[@]}; do
+        if grep -q "^\$d:" /etc/passwd ; then
+            echo "/etc/passwd has user \$d!"
+        else
+            echo "ERROR: /etc/passwd does not have user \$d"
+            res=1
+        fi
+      done
+
+      groups=("kubeadm-kas" "kubeadm-ks" "kubeadm-kcm" "kubeadm-etcd" "kubeadm-sa-key-readers")
+      for d in ${groups[@]}; do
+        if grep -q "^\$d:" /etc/group ; then
+            echo "/etc/group has user \$d!"
+        else
+            echo "ERROR: /etc/group does not have user \$d"
+            res=1
+        fi
+      done
+
+      # Here pgrep will return the PID of the process and we will pass the PID using xargs
+      # to the ps command as an argument.
```

Comment:
could you please add some more detail on what `ps o user:16 --no-headers -p` does?

Reply:
Done.